src/providers/connectedClusters.ts

```diff
@@ -0,0 +1,159 @@
+		return this.connectedClusters;
+	}
+
+	private updateK8SConfig(): void {
+		this.kubeConfig.loadFromDefault();
+		this.connectedClusters = this.kubeConfig.clusters.map(cluster => new ConnectedCluster(cluster));
+	}
+
+	private async getOcmResourceDefinitions(cluster: k8s.Cluster): Promise<OcmResourceDefinition[]> {
+		this.kubeConfig.setCurrentContext(cluster.name);
+		let k8sExtApi = this.kubeConfig.makeApiClient(k8s.ApiextensionsV1Api);
+		let apiResponse = await k8sExtApi.listCustomResourceDefinition();
+		if (apiResponse.response.statusCode !== 200) {
+			return Promise.resolve([]);
+		}
+		return Promise.resolve(
+			apiResponse.body.items
+			.filter(item => item.spec.group.includes('open-cluster-management'))
+			.map(item => new OcmResourceDefinition(item.spec)));
+	}
+
+	private async getOcmResources(spec: k8s.V1CustomResourceDefinitionSpec): Promise<OcmResource[]> {
+		let k8sCustomObjApi = this.kubeConfig.makeApiClient(k8s.CustomObjectsApi);
+
+		var listResourcesPromises: Promise<void>[] = [];
+		var customResources: OcmResource[] = [];
+
+		if (spec.scope === 'Namespaced') {
+			await this.getNamespacedResourceLists(spec, k8sCustomObjApi, listResourcesPromises, customResources);
+		} else {
+			await this.getClusterResourceLists(spec, k8sCustomObjApi, listResourcesPromises, customResources);
+		}
+
+		await Promise.allSettled(listResourcesPromises);
+		return Promise.resolve(customResources);
+	}
+
+	private async getNamespacedResourceLists(
+		spec: k8s.V1CustomResourceDefinitionSpec,
+		k8sCustomObjApi: k8s.CustomObjectsApi,
+		listResourcesPromises: Promise<void>[],
+		customResources: OcmResource[]): Promise<void> {
+
+		let k8sCoreApi = this.kubeConfig.makeApiClient(k8s.CoreV1Api);
+		let namespacesApiResponse = await k8sCoreApi.listNamespace();
+		if (namespacesApiResponse.response.statusCode === 200) {
+			namespacesApiResponse.body.items.forEach(ns => {
+				if (ns.metadata && ns.metadata.name) {
+					let namespace = ns.metadata.name;
+					spec.versions.forEach(async ver => {
+						listResourcesPromises.push(
+							k8sCustomObjApi.listNamespacedCustomObject(spec.group, ver.name, namespace, spec.names.plural)
+							.then(crApiResponse => {
+								if (crApiResponse.response.statusCode === 200) {
+									// @ts-ignore
+									crApiResponse.body.items.forEach(item =>
+										customResources.push(new OcmResource(item.metadata.name, namespace, ver.name)));
+								}
+							}));
+					});
+				}
+			});
+		}
+	}
+
+	private async getClusterResourceLists(
+		spec: k8s.V1CustomResourceDefinitionSpec,
+		k8sCustomObjApi: k8s.CustomObjectsApi,
+		listResourcesPromises: Promise<void>[],
+		customResources: OcmResource[]): Promise<void> {
+
+		spec.versions.forEach(async ver => {
+			listResourcesPromises.push(
+				k8sCustomObjApi.listClusterCustomObject(spec.group, ver.name, spec.names.plural)
+				.then(crApiResponse => {
+					if (crApiResponse.response.statusCode === 200) {
```

Comment:
error handling

Reply:
not for the demo, nither does testing.
for now, for error in k8s, an empty list is returned to not f-up the demo.